code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/AccountPolicy.java

```diff
@@ -23,6 +26,12 @@ public AccountPolicy(Account account, PolicyOptions options) {
 
     @Override
     public <T> T authorizeObject(T obj) {
+        ApiRequest request = ApiContext.getContext().getApiRequest();
+        @SuppressWarnings("unchecked")
+        Set<Object> whitelist = (Set<Object>) request.getAttribute("whitelist");
+        if(null != whitelist&& whitelist.contains(obj)) {
+            return obj;
+        }
```

Comment:
You're mixing things between DefaultPolicy and AccountPolicy.  A good indicator that something is wrong is that both classes need to know about the magical "whitelist" key.  Being that authorizeObject() in DefaultPolicy already returns the object, it would seem like grantObjectAccess should be a no-op in DefaultPolicy.  In AccountPolicy you can then have all the logic of grantObjectAccess and authorizeObject in the same place.